Automattic/wp-calypso
# Site Settings: abstract an HoC that redirects to alternative pages for non-JP/Atomic sites

This is an access management PR.

Currently, the ```redirectToGeneral()``` function is used both in ```ManageConnection``` and soon in the ```DisconnectFlow``` components. This PR abstracts this action into a separate HoC 
and makes the site redirection flexible to alternative urls.

Created in the context of (dis)connection, thus meant for non-Jetpack and Atomic sites, which should not have access to those actions. To be used in scenarios, where those access restrictions hold.

**To test**: 
Current functionality should not change. Hence, testing from https://github.com/Automattic/wp-calypso/pull/15716 applies here.
